tests/test_checks.py

```diff
@@ -93,3 +98,45 @@ def test_enum_type_boolean(tmpdir, rc):
                              |         | false
     ''')
     commands.check(context, manifest)
+
+
+def test_check_filename_incorrect_name(tmp_path: Path, rc: RawConfig):
+    create_tabular_manifest(tmp_path / 'hidrologija.csv', '''
+    d | r | b | m | property | type    | source
+    datasets/gov/example     |         |
+                             |         |
+      |   |   | Data         |         |
+      |   |   |   | value    | string  |
+    ''')
+
+    context = load_manifest_get_context(rc, tmp_path / 'hidrologija.csv', check_filename=True)
+
+    store = context.get('store')
+    manifest = store.manifest
+
+    with pytest.raises(InvalidFileName) as e:
+        commands.check(context, manifest)
+
+    assert e.value.message == (
+        f'Dataset namespace datasets/gov/example not match the csv filename {tmp_path}/hidrologija.'
+    )
+
+
+def test_check_filename_correct_name(tmp_path: Path, rc: RawConfig):
+    create_tabular_manifest(tmp_path / 'hidrologija.csv', '''
+    d | r | b | m | property | type    | source
+    datasets/gov/hidrologija |         |
+                             |         |
+      |   |   | Data         |         |
+      |   |   |   | value    | string  |
+    ''')
+
+    context = load_manifest_get_context(rc, tmp_path / 'hidrologija.csv')
```

Comment:
There are many tests without `check_filename`, but there is not tests with `check_filename=False`. We should try to avoid duplicating tests.